cmd/juju/commands/upgrademodel.go

```diff
@@ -833,6 +814,43 @@ func (c *baseUpgradeCommand) initVersions(
 	}, nil
 }
 
+// fetchStreamsVersions returns simplestreams agent metadata
+// for the specified stream. timeout ensures we don't block forever.
+func fetchStreamsVersions(
+	client toolsAPI, majorVersion int, stream string, timeout time.Duration,
```

Comment:
kinda wish we used `context.Context` more so timeouts and cancel propagation worked more idiomatically

Reply:
agreed, but the facade code does not support context. So we will still have to spawn a goroutine manually when we call the facade method.